koap/build.gradle.kts

```diff
@@ -53,3 +53,15 @@ npmPublishing {
         }
     }
 }
+
+tasks.register("run-js-validation-testing") {
+
+    description = "Runs the package for testing"
+    group = "Verification"
+    dependsOn("assemble")
+
+    doLast {
+        var response = runCommand("../gradlew -Pnpm.publish.repository.github.authToken=fakingit -Pnpm.publish.version=0.0.3-test24 assemble && cd koap_js_tests && npm uninstall @juullabs/koap && npm install file://../build/publications/npm/js && npm run test")
```

Comment:
You shouldn't need the `../gradlew -Pnpm.publish.repository.github.authToken=fakingit -Pnpm.publish.version=0.0.3-test24 assemble` portion, as your Gradle task is already depending on `assemble`, so it will execute before this task.

Reply:
I need a means to provide the option to set the published version to something. Without it, npm won't install it as it's version is "unspecified."

Reply:
For just running the tests, `unspecified` seems fine for a version. Unless that breaks other things?

Reply:
Without injecting a version, the npm install command fails with:

`> Task :koap:run-js-validation-testing FAILED
npm ERR! Invalid version: "unspecified"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/justin.tiffner/.npm/_logs/2020-11-17T18_08_03_817Z-debug.log`

Reply:
The npm package install will fail without a valid version number in the `package.json` file, so the command line version specifier is absolutely required. 

However as this depends on `assemble` there is no need to invoke it explicitly - it should be fine to just invoke this test task and provide the `-Pnpm.publish.version=<whatever>` argument to Gradle so that the npm assemble task has a version when it executes.

Reply:
Well, that's what I was getting at with my first response. I'm assuming I can't retroactively provide this argument to assemble here in the test task. This task dependsOn assemble, so won't be executed until after, right? 

Would "doFirst" execute before the dependsOn?

Reply:
> Would "doFirst" execute before the dependsOn?

No. The `doFirst/Last` closures are executed when the task runs. Anything a task `dependsOn` runs before the task. https://docs.gradle.org/current/dsl/org.gradle.api.DefaultTask.html#N1DB72